db/db_iter.cc

```diff
@@ -362,7 +362,7 @@ bool DBIter::FindNextUserEntryInternal(bool skipping_saved_key,
             break;
         }
       }
-    } else {
+    } else if (more_recent) {
```

Comment:
How about applying `if (more_recent)` just to `PERF_COUNTER_ADD(internal_recent_skipped_count, 1)` below?

Reply:
ok. Can you help double check the re-seek logic below still works?

Reply:
Sure. A unit test covering this reseek case when `timestamp_lb_` is specified can also help.